From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rong Bao <webmaster@csmantle.top>
Date: Sat, 3 Oct 2026 17:30:22 +0800
Subject: [PATCH 9/9] AOSCOS: Remove bloated optional dependencies from mach's
 UV env

These dependencies are bloatware, since:

1. They are marked as "pypi-optional".
2. They are mostly related to telemetry and acceleration, which are
   either unwanted or unnecessary for one-off compilations.
3. On platforms without pre-built wheels, like loongarch64, mips64 and
   riscv64, they are extremely slow to build.
---
 python/sites/mach.txt | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/python/sites/mach.txt b/python/sites/mach.txt
index 0f800b6813fe1c990f10a9bead37e274cbcdf3b1..42139c8e135862a8fbfb1375f445359fcfd6e49d 100644
--- a/python/sites/mach.txt
+++ b/python/sites/mach.txt
@@ -56,16 +56,16 @@ pth:tools/moztreedocs
 pth:xpcom/ds/tools
 pth:xpcom/geckoprocesstypes_generator
 pth:xpcom/idl-parser
-# glean-sdk may not be installable if a wheel isn't available
-# and it has to be built from source.
-pypi-optional:glean-sdk==70.2.1:telemetry will not be collected
-pypi-optional:orjson>=3.10:json operations will be slower in various tools
-# Mach gracefully handles the case where `psutil` is unavailable.
-# We aren't (yet) able to pin packages in automation, so we have to
-# support down to the oldest locally-installed version (5.4.2).
-pypi-optional:psutil>=5.4.2:telemetry will be missing some data
-pypi-optional:rtoml>=0.11.0:toml operations will be slower in various tools
-pypi-optional:zstandard>=0.11.1,<=0.25.0:zstd archives will not be possible to extract
+## glean-sdk may not be installable if a wheel isn't available
+## and it has to be built from source.
+#pypi-optional:glean-sdk==70.2.1:telemetry will not be collected
+#pypi-optional:orjson>=3.10:json operations will be slower in various tools
+## Mach gracefully handles the case where `psutil` is unavailable.
+## We aren't (yet) able to pin packages in automation, so we have to
+## support down to the oldest locally-installed version (5.4.2).
+#pypi-optional:psutil>=5.4.2:telemetry will be missing some data
+#pypi-optional:rtoml>=0.11.0:toml operations will be slower in various tools
+#pypi-optional:zstandard>=0.11.1,<=0.25.0:zstd archives will not be possible to extract
 vendored-fallback:pyyaml:third_party/python/pyyaml/lib/:faster native loading is disabled
 vendored:third_party/python/ansicon
 vendored:third_party/python/appdirs
